plugins/module_utils/raw.py

```diff
@@ -175,7 +176,7 @@ def execute_module(self):
                 continue
             kind = definition.get('kind', self.kind)
             api_version = definition.get('apiVersion', self.api_version)
-            if kind.endswith('List'):
+            if kind and kind.endswith('List'):
```

Comment:
I think if kind isn't defined here we're broken anyway, this should have been validated earlier. We could add an additional failout here if either kind or api_version aren't defined